File: frontend/src/services/leaveService.js
import apiClient from './apiClient'

export const leaveService = {
  applyLeave: (data) => apiClient.post('/leaves', data),
  getMyLeaves: () => apiClient.get('/leaves/my'),
  getTeamLeaves: () => apiClient.get('/leaves/team'),
  approveLeave: (id, note) => apiClient.patch(`/leaves/${id}/approve`, { note }),
  rejectLeave: (id, note) => apiClient.patch(`/leaves/${id}/reject`, { note }),
}

File: backend/src/main/java/com/timekeeper/service/LeaveService.java
package com.timekeeper.service;

import com.timekeeper.dto.request.CreateLeaveRequest;
import com.timekeeper.dto.request.LeaveActionRequest;
import com.timekeeper.dto.response.LeaveResponse;

import java.util.List;

public interface LeaveService {
    LeaveResponse applyLeave(String employeeId, CreateLeaveRequest request);
    List<LeaveResponse> getMyLeaves(String employeeId);
    List<LeaveResponse> getTeamLeaves(String managerId);
    LeaveResponse approveLeave(String leaveId, String approverId, LeaveActionRequest request);
    LeaveResponse rejectLeave(String leaveId, String approverId, LeaveActionRequest request);
}
